### DISCUSSION TITLE
Integrating PlanCraft Without Stripe and Creating Custom Plan Durations

### DISCUSSION THREAD

**Discussion by bulentsakarya** [realrashid/plan-craft#13] [Q&A]
Hello,

I'm trying to integrate the PlanCraft package into my system, but I won’t be using Stripe. I’ll be using one of the local payment systems.

When defining a plan:
```
"planId": {
  "monthly": "basic_price_monthly",
  "yearly": "basic_price_yearly"
}
```

Are these values specific to Stripe? I don’t fully understand them. Will they appear like this on the payment form? I've never integrated Stripe before.

Also, let’s say I want to create a plan for just 7 months:
```
PlanCraft::create('basic', 'Basic', '25', 'Monthly', [
  '10 Chirps',
  '5 Teams',
], [
  'max_chirps' => 2,
  'max_teams' => 5,
], 'basic_price_monthly')
->description('Basic plan users can create 10 Chirps per team and create 5 Teams.')
->trialDays(7);
```

If I use it like this, I'm creating a package with a 1-year cycle that only has a monthly payment option, with a fee of 25 units. However, I can't create a plan limited to 7 months, can I? Or is it possible to create a plan that charges every 2 years?

**Comment by realrashid** [comment]
Hello @bulentsakarya ,

1. **Plan ID Values**: The values you see in the `planId` definition (e.g., `"basic_price_monthly"` and `"basic_price_yearly"`) are intended as unique identifiers for your plans. They are not specific to Stripe and can be customized to fit your local payment system. These values are used internally to reference your plans when setting up billing. They won’t appear directly on the payment form but will help you manage your plans programmatically.

2. **Creating a 7-Month Plan**: Currently, the PlanCraft package does not support creating plans with a custom duration like 7 months directly out of the box. The billing cycles you define (monthly or yearly) correspond to recurring payments at those intervals. If you want to create a plan with a specific duration like 7 months, you would need to manage that outside of PlanCraft, perhaps by manually implementing a billing cycle or creating a one-time payment option.

3. **Billing for Longer Periods**: If you want to charge for a plan every 2 years, you can certainly set that up. You can define a plan with a yearly cycle and then handle the logic in your application to check for the 2-year interval for your specific requirements.

If you have any more questions or need further clarification, feel free to ask!

Happy Coding! 🚀
